packages/client/src/__tests__/or.test.ts

```diff
@@ -1,114 +0,0 @@
-import { getDMMF } from '@prisma/internals'
-import stripAnsi from 'strip-ansi'
-
-import { enums } from '../fixtures/enums'
-import { DMMFClass, makeDocument } from '../runtime'
-import { MergedExtensionsList } from '../runtime/core/extensions/MergedExtensionsList'
-
-describe('at least one validation', () => {
-  let dmmf
-  beforeAll(async () => {
-    dmmf = new DMMFClass(await getDMMF({ datamodel: enums }))
-  })
-
-  test('invalid or query', () => {
-    const select = {
-      where: {
-        OR: {
-          email: {},
-        },
-      },
-    }
-    const document = makeDocument({
-      dmmf,
-      select,
-      rootTypeName: 'query',
-      rootField: 'findManyUser',
-      extensions: MergedExtensionsList.empty(),
-    })
-    expect(String(document)).toMatchInlineSnapshot(`
-      query {
-        findManyUser(where: {
-          OR: [
-            {
-              email: {
-
-              }
-            }
-          ]
-        }) {
-          id
-          name
-          email
-          status
-          nicknames
-          permissions
-          favoriteTree
-          locationId
-          someFloats
-        }
-      }
-    `)
-    try {
-      document.validate(select, false, 'users')
-    } catch (e) {
-      expect(stripAnsi(e.message)).toMatchInlineSnapshot(`
-
-                                        Invalid \`prisma.users()\` invocation:
```

Comment:
Do we have this tested somewhere as a normal query that goes through the query engine, or on the engine side itself?

Reply:
Yes, we do have `query-validation` functional test and `applyValidationError` unit test.